src/components/menu/menu.test.js

```diff
@@ -1399,5 +1448,23 @@ context("Testing Menu component", () => {
           expect(callback).to.have.been.calledOnce;
         });
     });
+
+    it("should have correct keyboard navigation order when children of submenu update", () => {
+      CypressMountWithProviders(<MenuWithChildrenUpdating />);
+
+      submenu().eq(positionOfElement("first"), div).trigger("mouseover");
+      cy.wait(500);
```

Comment:
If we do want to verify the correct element is focused after each arrow press, what do you think about using aliases with a locator function instead of using `cy.wait()`? Similar to [this example from Cypress](https://docs.cypress.io/guides/references/best-practices#Unnecessary-wait-for-cyget)? This would mean the test wouldn't be dependent on the timeout duration and break if we change it.

Reply:
I'm not convinced this approach is necessary for this scenario tbh. The wait is just so the state can update with the setTimeout, it's entirely contrived for this test case rather than baked into the component being tested.